### PR TITLE
refactor(template): replace underscore render helpers

### PR DESCRIPTION
**Pull request by paulfalgout**
Related #241

## Linked issue

- Related #241

## What

- Resolve `templateContext` through Marionette's owned `getValue` helper.
- Serialize the framework-managed collection model array with native `Array.prototype.map` behavior.
- Add focused context receiver/read/error and collection order/reference/error coverage and regenerate affected main distribution artifacts.

## Why

Template Render only needs literal-key configuration lookup and mapping over Backbone's dense `Collection.models` array. Expressing those actual contracts directly removes its complete Underscore import without cloning unused generic behavior.

## Agent-development failure addressed

- Makes template-context resolution follow the same explicit configuration-value convention as the other migrated mixins.
- Makes collection serialization visibly depend on Backbone's maintained dense model array, with order and attributes-reference identity locked by tests.

## Public behavior contract

- Canonical behavior after this PR: resolve `templateContext` once with the view receiver and zero arguments; preserve one-sided identity and merge context over serialized data when both exist; serialize collection models in order to a new array containing their exact attributes objects.
- Superseded behavior removed, if any: direct Underscore `result` and `map` use in Template Render.
- Compatibility exception and removal condition, if any: arbitrary sparse arrays, array-like replacements, subclass species, shadowed `map`, and proxy-trap ordering are intentionally not reproduced because Backbone maintains `Collection.models` as a dense plain Array.

## Runtime cost

- [ ] Static or documentation only
- [ ] Development or test only; excluded from production entrypoints
- [x] Changes an existing production path
- [ ] Adds an explicitly opt-in runtime path

Measured impact or explanation:

- Bundle: the size/resource contract passes; direct production Underscore importers drop from 13 to 12. Brotli deltas versus base are ESM +10 B, CJS -12 B, UMD +3 B, and minified +1 B.
- Startup/hot path: no new startup work; collection serialization remains one linear pass with one attributes read per model.
- Allocations/retention: serialization still allocates exactly one result array and retains the existing attributes object references.

## Scope

- Affects `templateContext` mixing and default collection serialization only.
- Does not change model serialization, rendering, template invocation, content attachment, Backbone collection maintenance, or other Underscore consumers.

## Deployment Impact

- No website, package, tag, release, or deployment is performed by this PR.
- No consumer redeploy is required beyond adopting a future Marionette package containing this change.

## Validation

- [x] Tests cover the acceptance criteria and relevant edge cases.
- [x] Docs, examples, declarations, diagnostics, and rule metadata agree where relevant.
- [x] No private consumer, private fixture, or undocumented context is required.
- [x] I ran the commands listed below and am reporting their actual results.

Commands:

```sh
npm ci --no-audit --no-fund
npm exec vitest -- run test/unit/mixins/template-render.spec.js test/unit/utils/get-value.spec.js --reporter=dot
npm run lint:ci
npm run build
npm test -- --reporter=dot
npm run test:source
npm run test:dist
npm run test:fixtures
npm run size
npm run test:performance
npm run performance:timing
git diff --check
```

Results: focused tests passed 34/34; full tests passed 1,454/1,454 across 76 files; source, distribution, packed fixture, lint, build, size/resource, 92 performance-contract, hosted timing, and diff validation passed. Independent exact-head review passed 104/104 focused/integration assertions with no findings.

## Package and browser impact

- Entry points or install fixtures affected: main ESM, CJS, UMD, and minified artifacts are regenerated; packed ESM/CJS/type/Vite fixtures pass.
- Browser contracts affected: none; supported environments already provide native Array mapping.
- Production/dev/test module-graph impact: Template Render stops importing Underscore; other production consumers still retain it.

## Rollback or deprecation

- Revert this single commit and regenerate the main distribution artifacts; no data migration or compatibility bridge is involved.

## Review notes

- Native mapping is deliberately scoped to the supported Backbone-managed dense plain models array rather than emulating Underscore behavior for corrupted or replaced internals.
- Tests assert order, new-array identity, strict attributes-reference identity, one read per visited model, and first-error cutoff.

---
## Summary by cubic
Replaces Underscore's `result` and `map` calls in `mixins/template-render.js` with Marionette's `getValue` helper and native `Array.prototype.map`, dropping the direct Underscore import from the template render path without changing behavior.

- `templateContext` resolution now uses `getValue` with the view as receiver and zero arguments, matching the convention used by other migrated mixins.
- Collection serialization maps over Backbone's dense `collection.models` array; new tests lock order, new-array identity, and attributes-reference identity.
- Adds tests for context receiver/read/error behavior and collection serialization details.
- Bundle size impact is minimal: ESM +10 B, CJS -12 B, UMD +3 B, minified +1 B.

<sup>Written for commit cf0a37c69528882ce6f162eed50e5a97a1b709a9. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/marionettejs/marionette/pull/281?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

**Comment by paulfalgout** [comment]
## Code-review approval — exact head `62bbadf0`

**Approved.** CI green, +2 B, 17,334 remaining. Production Underscore importers **14 → 13**.

`result` → `getValue` is mechanical and equivalent per #278.

**`_.map` → native `.map` is a real narrowing, and I think it's worth one line of documentation rather than leaving it in the PR body.** The three divergences, all in degenerate-input territory:

```
collection.models is…        master (_.map)          this PR (native .map)
undefined                    returns []              TypeError: cannot read 'map' of undefined
array-like, not an Array     works                   TypeError: .map is not a function
sparse (has holes)           throws on the hole      skips holes, holes in result
```

The sparse case is arguably an improvement — `_.map` calls the iteratee for holes, so `model.attributes` blows up on `undefined`, where native `.map` skips them.

The first two matter because `serializeData` guards with a bare truthiness check:

```js
if (this.collection) { return { items: this.serializeCollection() }; }
```

Nothing requires `models` to be a dense `Array`. For a real `Backbone.Collection` it always is, so this is safe for the documented input — and your description says exactly that. But `serializeCollection` is a documented override point, ROADMAP line 430 puts "validating optional Backbone integration" in scope for #241, and a collection-like object with an array-like `models` is precisely the shape that becomes plausible as Backbone moves toward optional. Today it degrades to `[]`; after this it throws.

I'd suggest either a sentence in the reference stating `collection.models` must be a dense `Array`, or a migration row — whichever fits. Not a blocker: the change is correct for every supported input, and anyone hitting it gets a loud failure rather than a silent one.

Still source-only review; `node_modules` remains empty.

<sub>Reviewed with Claude Code.</sub>

File: test/unit/mixins/template-render.spec.js [hunk 1]
```diff
@@ -242,6 +255,39 @@ describe('template-render', function() {
           .to.be.calledOnce
           .and.calledWith(renderer.template, { items: [{ id: 1 },{ id: 2 }] });
       });
+
+      it('preserves model order and attribute object identity', function() {
+        const serialized = renderer.serializeCollection();
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply test/unit/mixins/template-render.spec.js:260]: Handled in `test/unit/mixins/template-render.spec.js` with an explicit strict non-identity assertion against `collection.models`.

**Comment by paulfalgout** [comment]
**Approval carries** to `cf0a37c6` (the SHA I fetched). Tests-only — `dist/`, `mixins/`, `modules/`, and `utils/` are byte-identical to `62bbadf0`. CI green.

`expect(serialized).to.not.equal(collection.models)` is a good addition: both `_.map` and native `.map` return a fresh array, and pinning that stops a future "optimization" from returning `collection.models` directly, which would hand template code a live reference to the collection's internal array.

My documentation note from the previous pass is untouched — no `docs/` changes on the branch — and I'll leave it there rather than repeat it. It was non-blocking and the call on whether `collection.models` must be a dense `Array` is a contract decision rather than a code defect.

<sub>Reviewed with Claude Code.</sub>
